pkg/kubectl/cmd/create_podsecuritypolicy.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	pspv1beta1 "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgo "k8s.io/client-go/kubernetes/typed/extensions/v1beta1"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/util/i18n"
+)
+
+var (
+	PSPLong = templates.LongDesc(i18n.T(`
+		Create a PodSecurityPolicy.`))
+
+	pspExample = templates.Examples(i18n.T(`
+
+		# Create a PodSecurityPolicy named "example" that simply uses default values.
+		kubectl create podsecuritypolicy example
+
+		# Create a PodSecurityPolicy named "my-psp"
+		kubectl create podsecuritypolicy privileged --selinux=MustRunAs
+		`))
+)
+
+type CreatePSPOptions struct {
+	Name               string
+	SELinux            string
+	RunAsUser          string
+	SupplementalGroups string
+	FSGroup            string
+
+	DryRun       bool
+	OutputFormat string
+	Client       clientgo.ExtensionsV1beta1Interface
+	Mapper       meta.RESTMapper
+	Out          io.Writer
+	PrintObject  func(obj runtime.Object) error
+}
+
+func NewCmdCreatePSP(f cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	c := &CreatePSPOptions{
+		Out: cmdOut,
+	}
+	cmd := &cobra.Command{
+		Use: "podsecuritypolicy NAME --selinux=rule --supplemental-groups=rule --run-as-user=rule --fs-group=rule [other options] [--dry-run]",
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"psp"},
+		Short:                 PSPLong,
+		Long:                  PSPLong,
+		Example:               pspExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(c.Complete(f, cmd, args))
+			cmdutil.CheckErr(c.Validate())
+			cmdutil.CheckErr(c.RunCreatePSP())
+		},
+	}
+
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddPrinterFlags(cmd)
+	cmdutil.AddDryRunFlag(cmd)
+	cmd.Flags().String("selinux", "RunAsAny", "define the strategy that will dictate the allowable labels that may be set")
+	cmd.Flags().String("run-as-user", "RunAsAny", "control what user ID containers can run as.")
+	cmd.Flags().String("supplemental-groups", "RunAsAny", "control which group IDs containers can add.")
+	cmd.Flags().String("fs-group", "RunAsAny", "provide a whitelist of allowed volume types.")
```

Comment:
> provide a whitelist of allowed volume types

No, it isn't.

Reply:
I think that in order to improve UX, we should mention valid values in the help output.